test/sub_client/test_client.py

```diff
@@ -360,6 +360,31 @@ async def test_client_protocol_handler():
     assert result == reply
 
 
+async def test_client_broadcast_echo():
+    """Test the client protocol handles broadcasts with local echos correctly"""
```

Comment:
Local echo, is also used in the server.

Reply:
Can a server not use a usb adapter for serial communication ?
so local echo can happen there too.